### PR TITLE
Adding sections for new terraform config

### PR DESCRIPTION
**Pull request by EMcWhinn**
Create product docs for terraform-provider-aap

https://issues.redhat.com/browse/ACA-1187

Affects `controller-user-guide`

### CODE REVIEW

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+[id="proc-controller-inv-source-terraform"]
+
```
[the remark this answers is not in the document]
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:2]: Updated

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+= Terraform State
+
+Configure an inventory sourced from a Terraform State file.
+
+.Procedure
+
+. From the navigation panel, select {MenuAEProjects}.
+. On the *Projects* page, click btn:[Create project] to start the *Create Project* window.
+** Enter the appropriate details according to the steps in xref:proc-controller-adding-a-project[Adding a new project].
+** Use the configuration files at link:https://github.com/hakbailey/terraform-aap-demo[terraform-aap-demo].
```
Comment (hakbailey) [inline downstream/modules/platform/proc-controller-inv-source-terraform.adoc:14]: We probably don't want to link to my personal repo here. Which configuration files are being referenced?

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+This should match the remote backend configured in the *Terraform backend credential*.
+The following is an example s3 backend:
++
+----
+backend_type: s3
+----
++
+//The current 2.5 test environment does not have the following option yet:
+. Select an *Execution Environment* that has a Terraform binary. 
+This is required for the inventory plugin to run the Terraform commands that read inventory data from the Terraform state file.
```
Comment (hakbailey) [inline downstream/modules/platform/proc-controller-inv-source-terraform.adoc:37]: We could optionally link here to the example terraform EE configuration repo: https://github.com/ansible-cloud/terraform_ee

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc (hunk 2 above)
Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:14]: Removed this sub-step.

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+.Additional resources
+For more information, see the link:https://github.com/ansible-cloud/terraform_ee[Terraform EE] readme that has an example {ExecEnvShort} configuration with a Terraform binary.
+
+== Terraform provider for {PlatformNameShort}
```
Comment (EMcWhinn) [inline downstream/modules/platform/proc-controller-inv-source-terraform.adoc:43]: @hakbailey I've added a sub-section here for the Terraform provider, does this work? And are there steps I can grab from somewhere to document here?

Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:43]: Got it, thanks @hakbailey.

Comment (gravesm) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:43]: I believe we should have docs we can link to on the Hashi registry in ~2 weeks. I'm not sure what the timing looks like for docs publishing for AAP. If possible, I think pausing this for a couple weeks until we have the official provider docs published that we can link to would be best.

Comment (EMcWhinn) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:43]: Thanks @gravesm, docs hardening is June 4th - June 18th (2 wks) with publishing on June 18th, so ~2 weeks to confirm the docs here should be fine.

Comment (hakbailey) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:43]: Provider has been released! Here's the official docs link: https://registry.terraform.io/providers/ansible/aap/latest/docs

File: downstream/modules/platform/proc-controller-inv-source-terraform.adoc [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Inventories created this way are managed by Terraform and you must not edit them in {PlatformNameShort} as it can introduce drift to the Terraform deployment.
+
+You can create inventories and hosts within the Terraform configuration by using the Terraform provider for {PlatformNameShort}.
+For more information, see the link:https://registry.terraform.io/providers/ansible/aap/latest/docs[AAP Provider] section of the Terraform documentation.
```
Comment (EMcWhinn) [inline downstream/modules/platform/proc-controller-inv-source-terraform.adoc:48]: Great thanks @hakbailey, I've added the info. Does it all look good to you?

Comment (hakbailey) [reply downstream/modules/platform/proc-controller-inv-source-terraform.adoc:48]: Looks great!

**Review by hakbailey — APPROVED** [review]

**Review by ianf77 — APPROVED** [review]
LGTM

**Pull request merged**